pkg/testing/cmd_test.go

```diff
@@ -0,0 +1,189 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testing
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"gopkg.in/check.v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/kanisterio/kanister/pkg/kopia/command"
+	"github.com/kanisterio/kanister/pkg/kopia/command/storage"
+	"github.com/kanisterio/kanister/pkg/kopia/repository"
+	"github.com/kanisterio/kanister/pkg/kube"
+	"github.com/kanisterio/kanister/pkg/secrets"
+	"github.com/kanisterio/kanister/pkg/testutil"
+)
+
+const (
+	testPodName = "kopia-cmd-"
+)
+
+type KopiaCmdSuite struct {
+	cli       kubernetes.Interface
+	namespace string
+	locType   storage.LocType
+}
+
+var _ = check.Suite(&KopiaCmdSuite{locType: storage.LocTypeS3})
+var _ = check.Suite(&KopiaCmdSuite{locType: storage.LocTypeFilestore})
+
+func (s *KopiaCmdSuite) SetUpSuite(c *check.C) {
+	config, err := kube.LoadConfig()
+	c.Assert(err, check.IsNil)
+	cli, err := kubernetes.NewForConfig(config)
+	c.Assert(err, check.IsNil)
+	s.cli = cli
+
+	ctx := context.Background()
+	ns := testutil.NewTestNamespace()
+	ns.GenerateName = "kanister-datatest-"
+
+	cns, err := s.cli.CoreV1().Namespaces().Create(ctx, ns, metav1.CreateOptions{})
+	c.Assert(err, check.IsNil)
+	s.namespace = cns.GetName()
+	sa := &v1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-sa-",
+			Namespace:    s.namespace,
+		},
+	}
+
+	sa, err = cli.CoreV1().ServiceAccounts(s.namespace).Create(ctx, sa, metav1.CreateOptions{})
+	c.Assert(err, check.IsNil)
+	os.Setenv("POD_NAMESPACE", s.namespace)
+	os.Setenv("POD_SERVICE_ACCOUNT", sa.Name)
+}
+
+func (s *KopiaCmdSuite) TearDownSuite(c *check.C) {
+	os.Unsetenv("POD_NAMESPACE")
+	os.Unsetenv("POD_SERVICE_ACCOUNT")
+	ctx := context.Background()
+	if s.namespace != "" {
+		_ = s.cli.CoreV1().Namespaces().Delete(ctx, s.namespace, metav1.DeleteOptions{})
+	}
+}
+
+func (s *KopiaCmdSuite) TestRepositoryCreate(c *check.C) {
+	locSecret, credSecret := s.createLocationAndCredSecrets(c)
+	pod := s.startKanisterToolsPod(c, locSecret, credSecret)
+	err := repository.CreateKopiaRepository(s.cli, s.namespace, pod.Name, pod.Spec.Containers[0].Name, command.RepositoryCommandArgs{
+		CommandArgs: &command.CommandArgs{
+			RepoPassword:   "test-pass123",
+			ConfigFilePath: "/tmp/config",
+			LogDirectory:   "/tmp/logs",
+		},
+		Hostname:        "test-hostname",
+		Username:        "test-username",
+		RepoPathPrefix:  fmt.Sprintf("test-path/test-%v", time.Now().Unix()),
+		ContentCacheMB:  0,
+		MetadataCacheMB: 0,
+		CacheDirectory:  "/tmp/cache",
+		Location:        locSecret.Data,
+	})
+	c.Assert(err, check.IsNil)
```

Comment:
Just to add an extra validation, Is there a way to validate the status of the repository after creation?

Reply:
Yes, we can run `repository status` command to check the status. Also, we can run `repository connect` to verify repository is created using expected credentials only. Will add those steps as well once the parent PRs are approved.